homeassistant/components/cert_expiry/config_flow.py

```diff
@@ -56,15 +56,31 @@ def _prt_in_configuration_exists(self, user_input) -> bool:
         except socket.timeout:
             _LOGGER.error("Timed out connecting to %s", host)
             self._errors[CONF_HOST] = "connection_timeout"
+        except ConnectionRefusedError:
+            _LOGGER.error("Connection refused: %s", host)
+            self._errors[CONF_HOST] = "connection_refused"
         except ssl.CertificateError as err:
-            if "doesn't match" in err.args[0]:
+            if "Hostname mismatch" in err.verify_message:
                 _LOGGER.error("Certificate does not match host: %s", host)
                 self._errors[CONF_HOST] = "wrong_host"
+            elif "certificate has expired" in err.verify_message:
+                _LOGGER.error("Certificate has expired: %s", host)
```

Comment:
Why is this an error for setting up the config flow? Isn't the whole reason this sensor exists to check this?

Reply:
The error is meant to interactively help fix an existing broken setup. We could create a new sensor and assume the user will check the logs for more info on why the sensor is reporting a failed cert, but presenting this info in the UI feels like it would help resolve the problem faster.

Reply:
But what if the user wants to set up the sensor, see that it's invalid, then fix the sensor and see it jump to valid ?

Reply:
You wouldn't prevent a leak detector from being paired if it was currently detecting water.

Reply:
@balloob : I'm not so sure... this component give us the number of days before expiration.  If the certificate is already expired, the state will be negative and I'm not sure it will really work.

I would agree with you if the state was "IsExpired: yes/no" and one of the attribute was the number of days to go to the expiration.

Reply:
After some thought I agree with @balloob and this will be allowed in a new PR.